checker/tests/resourceleak-arrays/patternMatchOwningArrayLoops.java

```diff
@@ -0,0 +1,258 @@
+import java.net.Socket;
+import org.checkerframework.checker.mustcall.qual.Owning;
+import org.checkerframework.checker.mustcallonelements.qual.OwningArray;
+
+class PatternMatchOwningArrayLoops {
+  private final int n = 10;
+  private final String myHost = "";
+  private final int myPort = 1;
+  // :: error: owning.array
+  @Owning Socket[] s;
+
+  // @OwningArray non-(1dArray/collection) is not allowed
+  public void owningArrayNonArray() {
+    // :: error: owningarray.nonarray
+    @OwningArray Socket s;
+    // :: error: owningarray.nonarray
+    @OwningArray Socket[][] sMultiDimensional;
+  }
+
+  public void illegalOwningArrayAssignment() {
+    Socket[] s = new Socket[n];
+    // this is a false positive, but we only allow assignment
+    // of an @OwningArray to a new [].
```

Comment:
We could consider using a small type system (or just a dataflow analysis...) to permit this case. Effectively, we need an abstract value for "newly-created array". The CF's local dataflow analysis would be able to propagate that forward for us, if we could express it.

(To be clear, this is probably not high-enough priority to be worth actually doing. But, this false positive is definitely solvable and is the sort of FP that _will_ frustrate users.)